src/engine_core/src/input/InputManager.hpp

```diff
@@ -5,18 +5,67 @@
 */
 
 #pragma once
-#include "KeyData.hpp"
+#include "definitions/KeyData.hpp"
+#include "definitions/MouseButton.hpp"
+#include "definitions/MouseData.hpp"
+#include <glm/vec2.hpp>
 #include <map>
 
 class InputManager
```

Comment:
Can we make it a non-static class and pass it to other systems through some kind of dependency injection?

Not using global state make it easier to test